src/sysinfo/nvapi_adapter.cpp

```diff
@@ -147,7 +157,51 @@ namespace dxvk {
         return 0;
     }
 
-    bool NvapiAdapter::isVkDeviceExtensionSupported(const std::string name) { // NOLINT(performance-unnecessary-value-param)
+    bool NvapiAdapter::GetLUID(LUID *luid) const {
+        if (m_deviceIdProperties.deviceLUIDValid) {
+            memcpy(luid, &m_deviceIdProperties.deviceLUID, sizeof(*luid));
+            return true;
+        }
+        return false;
+    }
+
+    bool NvapiAdapter::GetArchitectureId(NV_GPU_ARCHITECTURE_ID *archId) const {
+        // This function only makes sense on NVIDIA GPUs, bail out early if a
+        // non-NVIDIA GPU is detected.
+        if (m_deviceProperties.vendorID != 0x10de)
+            return false;
+
+        // Start at Kepler and update as features are detected
+        *archId = NV_GPU_ARCHITECTURE_GK100;
```

Comment:
We could do this the other way around, starting by testing for Ampere, return corresponding architecture when matching or performing the next test. That way we could skip some test. Thus with above in mind something like: 

```c++
// ...
if (isVkDeviceExtensionSupported(VK_KHR_FRAGMENT_SHADING_RATE_EXTENSION_NAME)
  && m_deviceFragmentShadingRateProperties.primitiveFragmentShadingRateWithMultipleViewports)
  return NV_GPU_ARCHITECTURE_GA100;

// ...
if (isVkDeviceExtensionSupported(VK_NV_SHADING_RATE_IMAGE_EXTENSION_NAME))
  return NV_GPU_ARCHITECTURE_TU100;

...
```

Reply:
I think when we initialise the nvapi, we already go over the adapters and can extract all the useful information during the initalisation. If there are no strict requirements to the duration of such, I think, we can just do these checks once there, and here only have a single if and return. Now, every single call to this function will have consequences of many nested calls to vulkan.

Reply:
> I think when we initialise the nvapi, we already go over the adapters and can extract all the useful information during the initalisation, that was what I did in my fork. If there are no strict requirements for the duration of such, I think, we can just do these checks once there, and here only have a single if and return. Now, every single call to this function will have consequences of many nested calls to vulkan.

No, this is fine here and no different than e.g. determining the memory size. We already have all the needed information/structures from Vulkan on "our" side (`vulkan-1.dll` is already unloaded here).

Reply:
Sure. I've changed this in the next revision.